### PR TITLE
Editing the mentor profile page and enhancements to the application

### PR DESCRIPTION
**Pull request by aniarya82**

### CODE REVIEW

**Comment by icyflame** [comment]
- Please add the gem `pg` back to the gem file.

File: app/views/student/show.html.erb [hunk 1]
```diff
@@ -8,83 +23,88 @@
...
+       </div>
+
+       <a href="#" class="btn btn-primary" id="send" role="button" style="border-color:black ">SEND</a> -->
+     </div>
+    
+
+      <div class="row" style="margin:5px 4px 5px 4px">
+        <h1>Recent updates </h1>
+        <div class="well">
+          <img src="det_pic.jpg" alt="...">
```
Comment (icyflame) [inline app/views/student/show.html.erb]: Please refactor the path of the images in this file. 

All the images must have a `src` attribute from `/assets/images/imagename.jpg`

File: app/views/students/registrations/_register.html.erb [hunk 2]
```diff
@@ -17,16 +17,37 @@
...
+  <div class="field">
+    <%= f.label :preference3 %>
+    <%= f.text_field :preference3 %>
+  </div>
 </div>
 
 <div class="field">
-  <%= f.label :joining_year %><br />
-  <%= f.text_field :joinyear %>
+  <%= f.label :current_academic_year %><br />
```
Comment (icyflame) [inline app/views/students/registrations/_register.html.erb:49]: Nice thinking!

But please add the joining year field apart from them urgent year attribute. 

We need to know both the year that the student joined (`joinyear`) and the year in which he is currently studying(`current_year`)

File: db/migrate/20150226164000_add_preference_to_student.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddPreferenceToStudent < ActiveRecord::Migration
```
Comment (icyflame) [inline db/migrate/20150226164000_add_preference_to_student.rb]: We don't need this field.

We will have one `preference` field for mentor and 3 preference fields (as defined by you after this) for the student. 

So, please remove this migration and rebuild the schema and push to this branch.

**Comment by icyflame** [comment]
@aniarya82 good work! Please check the line notes above and modify as required! 

Preferably, please do this in separate commits.

File: app/views/students/registrations/_register.html.erb (hunk 2 above)
Comment (aniarya82) [inline app/views/students/registrations/_register.html.erb:49]: but we can know the join year from the roll number itself

Comment (icyflame) [inline app/views/students/registrations/_register.html.erb]: Yes. Indeed! Thanks for that!

Comment (icyflame) [inline app/views/students/registrations/_register.html.erb:49]: @aniarya82 Actually, let the joining year field be there. We may need it later. And parsing it out of the roll number string can be unnecessarily tough. 

So, add both the joining year and the current academic year fields. 

And also, the current academic year field needs to be a drop down. Can you do that or should I do it after the code is merged?

Best Regards,

Siddharth Kannan,
Sophomore, Dept. of Mechanical Engg,
Indian Institute of Technology, Kharagpur.

**Comment by icyflame** [comment]
- Please do not close this without discussion. Resolve the notes mentioned above. 
- Also, commit the changed `Gemfile.lock`. After changing Gemfile, run `bundle` so that Gemfile.lock will get updated automatically. Commit that file too. 
- change the name of the PR to something more appropriate

File: db/schema.rb [hunk 4]
```diff
@@ -98,6 +99,11 @@
     t.string   "last_sign_in_ip"
     t.string   "phone"
     t.string   "skypename"
+    t.text     "preference"
```
Comment (icyflame) [inline db/schema.rb]: Although you have removed the preference migration, the schema has not been updated yet. Please run `rake db:migrate` and then commit the schema file again.

**Comment by icyflame** [comment]
- Commit the changed Gemfile.lock file. That was mentioned in the above comment. Run `bundle`. This will update the `Gemfile.lock` file. After that, stage this file and commit it.

File: app/views/student/show.html.erb [hunk 5]
```diff
@@ -1,6 +1,21 @@
-<br/>
+<h1>Student Profile Page: ID <%= params[:id] %> </h1>
```
Comment (icyflame) [inline app/views/student/show.html.erb:1]: Please reset this file to it's old state. The profile of the mentee has been perfected, and the changes you have proposed here will only regress this back to the way it was before my last set of changes.

It is the three column profile page, with news feed and everything else, built in.

File: app/views/students/registrations/_register.html.erb [hunk 6]
```diff
@@ -17,15 +17,41 @@
   </select>
 </div>
 
-<div class="field">
+<!-- <div class="field">
```
Comment (icyflame) [inline app/views/students/registrations/_register.html.erb]: Please do not comment this out. We need to take the hall of the student, when he/she fills up the registration form.

**Comment by icyflame** [comment]
# Phase 1 Debugging Complete
- correct `Gemfile.lock` could not be committed
- `schema.rb` was edited manually, instead of auto-generating it.
- the `academic year` field must be a select drop box.

**Comment by icyflame** [comment]
# Phase 2: Visual Debugging Start

**Comment by icyflame** [comment]
image
- The data is shown twice on the contacts page
- Please edit the contact page accordingly.

**Comment by icyflame** [comment]
Merged this in the squashed commit 5ffaf52a6523a72fd97b64a2f74a15c80d2099ad.

**Pull request closed without merging**